src/Stryker.Core/Stryker.Core/Mutators/MathMutator.cs

```diff
@@ -0,0 +1,146 @@
+            [MathExpression.Asinh] = new[] { MathExpression.Asin, MathExpression.Acosh, MathExpression.Atanh },
+            [MathExpression.Atan] = new[] { MathExpression.Atanh, MathExpression.Acos, MathExpression.Asin },
+            [MathExpression.Atanh] = new[] { MathExpression.Atan, MathExpression.Acosh, MathExpression.Asinh },
+            [MathExpression.BitDecrement] = new[] { MathExpression.BitIncrement },
+            [MathExpression.BitIncrement] = new[] { MathExpression.BitDecrement },
+            [MathExpression.Ceiling] = new[] { MathExpression.Floor },
+            [MathExpression.Cos] = new[] { MathExpression.Cosh, MathExpression.Sin, MathExpression.Tan },
+            [MathExpression.Cosh] = new[] { MathExpression.Cos, MathExpression.Sinh, MathExpression.Tanh },
+            [MathExpression.Exp] = new[] { MathExpression.Log },
+            [MathExpression.Floor] = new[] { MathExpression.Ceiling },
+            [MathExpression.Log] = new[] { MathExpression.Exp, MathExpression.Pow },
+            [MathExpression.MaxMagnitude] = new[] { MathExpression.MinMagnitude },
+            [MathExpression.MinMagnitude] = new[] { MathExpression.MaxMagnitude },
+            [MathExpression.Pow] = new[] { MathExpression.Log },
+            [MathExpression.ReciprocalEstimate] = new[] { MathExpression.ReciprocalSqrtEstimate },
+            [MathExpression.ReciprocalSqrtEstimate] = new[] { MathExpression.ReciprocalEstimate, MathExpression.Sqrt },
+            [MathExpression.Sin] = new[] { MathExpression.Sinh, MathExpression.Cos, MathExpression.Tan },
+            [MathExpression.Sinh] = new[] { MathExpression.Sin, MathExpression.Cosh, MathExpression.Tanh },
+            [MathExpression.Tan] = new[] { MathExpression.Tanh, MathExpression.Cos, MathExpression.Sin },
+            [MathExpression.Tanh] = new[] { MathExpression.Tan, MathExpression.Cosh, MathExpression.Sinh }
+        };
+
+        /// <summary> Apply mutations to an <see cref="InvocationExpressionSyntax"/> </summary>
+        public override IEnumerable<Mutation> ApplyMutations(ExpressionSyntax node)
+        {
+            var original = node;
+            if (node.Parent is ConditionalAccessExpressionSyntax || node.Parent is MemberAccessExpressionSyntax)
+            {
+                yield break;
+            }
+
+            foreach (var mutation in FindMutableMethodCalls(node, original))
+            {
+                yield return mutation;
+            }
+        }
+
+        private static IEnumerable<Mutation> FindMutableMethodCalls(ExpressionSyntax node, ExpressionSyntax original)
+        {
+            while (node is ConditionalAccessExpressionSyntax conditional)
+            {
+                foreach (var subMutants in FindMutableMethodCalls(conditional.Expression, original))
+                {
+                    yield return subMutants;
+                }
+                node = conditional.WhenNotNull;
+            }
+
+            for (; ; )
```

Comment:
First, that's an obscure way to create an infinite loop from the times of C, I'd prefer any clearer form, e.g. `while (true)`.

Second, I haven't created mutators myself but I don't think I've seen infinite loops in the mutators' code (@dupdob correct me if I am wrong) and so I'm just generally paranoid about them. Can you somehow verbally describe this need here? Or maybe extract a method and give it a very expressive name :)

Reply:
Hmm, alright, good to know! But my feedback (and I am just a random dev) is the same - the code could be clearer here, we don't need to be uniform in crypticity ;)

Reply:
Yes, sure, I will think about how to refactor this place. Then, maybe, the good idea will be to refactor the same place in `LinqMutator` in the same way.

Reply:
I understand why you picked `LinqMutator` as a template, but it is needlessly complex for Math methods: as those are static methods, you do not need to worry about conditional accessor or chained invocation. You also don't need to worry about MemberBinding expression.
So you need to look for invocation looking like `Math.[Sin|Cos|Tan|...]`, which are `InvocationExpression` using  a `SimpleMemberAcessExpression` as its Expression. This one has an `IdentifierName` (token 'Math') as an expression and another `IndentifierName`(identifier 'Sin', 'Cos', ...).

Therefore you can drop the initial `while` loop and transform the switch case to a simpler `if`.

You can use sharplab.io to have an idea of how different syntax constructs are represented with Roslyn.
Here is a [simple example](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLACgUGYACdQgYUIG9dCbiiUAWQgWQAoBKS62n5sYABYA6AMoBLAHasMABnYBubjQC+uZUA==).
I think you will find this useful.

Reply:
Thank you. I will change the implementation to use this way.

Reply:
Anyway, code is much clearer now, good job!